app/repos/cdo.js

```diff
@@ -367,88 +368,204 @@ const getSortOrder = (sort) => {
+ * @type {boolean} [nonComplianceLetterSent]
+ */
 /**
  * @typedef GetSummaryCdos
- * @param {{ status?: CdoStatus[]; withinDays?: number; nonComplianceLetterSent?: boolean }} [filter]
+ * @param {SummaryCdoFilter} [filter]
  * @param {CdoSort} [sort]
- * @return {Promise<SummaryCdo[]>}
+ * @return {Promise<{ count: number; cdos: SummaryCdo[] }>}
  */
 
 /**
- * @type {GetSummaryCdos}
+ * @param {SummaryCdoFilter} [filter]
+ * @param {CdoSort} [sort]
+ * @param cache
+ * @return {Promise<{ count: number; cdos: SummaryCdo[] }>}
  */
-const getSummaryCdos = async (filter, sort) => {
+const getSummaryCdos = async (filter, sort, cache) => {
   const where = {}
 
   if (filter.status) {
-    const statusArray = filter.status.map(status => statuses[status])
-    where['$status.status$'] = statusArray
+    where['$status.status$'] = cdoStatusFilter(filter.status)
   }
 
   if (filter.withinDays) {
-    const day = 24 * 60 * 60 * 1000
-    const withinMilliseconds = filter.withinDays * day
-    const now = new Date()
-    now.setUTCHours(0, 0, 0, 0)
-    const withinDaysDate = new Date(now.getTime() + withinMilliseconds)
-
-    where['$registration.cdo_expiry$'] = {
-      [Op.lte]: withinDaysDate
-    }
+    where['$registration.cdo_expiry$'] = cdoExpiryFilter(filter.withinDays)
   }
 
   if (filter.nonComplianceLetterSent !== undefined) {
-    const operation = filter.nonComplianceLetterSent === false ? Op.is : Op.not
+    where['$registration.non_compliance_letter_sent$'] = cdoNonComplianceFilter(filter.nonComplianceLetterSent)
+  }
 
-    where['$registration.non_compliance_letter_sent$'] = {
-      [operation]: null
-    }
+  const genericQueryFields = {
+    where,
+    include: summaryCdoInclude()
   }
 
   const order = getSortOrder(sort)
 
   const cdos = await sequelize.models.dog.findAll({
+    ...genericQueryFields,
     attributes: [
       'id', 'index_number', 'status_id',
       [sequelize.fn('COALESCE', sequelize.col(policeForceCol), 'ZZZZZZ'), 'police_force_aggregrate']
     ],
-    where,
-    include: [
-      {
-        model: sequelize.models.registered_person,
-        as: 'registered_person',
-        attributes: { exclude: ['created_at', 'updated_at', 'deleted_at', 'person_id', 'dog_id', 'person_type_id'] },
-        include: [{
-          model: sequelize.models.person,
-          as: 'person',
-          attributes: ['id', 'first_name', 'last_name', 'person_reference']
-        }]
-      },
-      {
-        model: sequelize.models.status,
-        as: 'status'
-      },
-      {
-        model: sequelize.models.registration,
-        as: 'registration',
-        attributes: ['id', 'cdo_expiry', 'joined_exemption_scheme', 'non_compliance_letter_sent'],
-        include: [
-          {
-            model: sequelize.models.police_force,
-            as: 'police_force',
-            paranoid: false
-          }
-        ]
-      }
-    ],
     order
   })
 
-  return cdos
+  const count = await sequelize.models.dog.count({
```

Comment:
I don;t think you need to do a count here as you could just check the length of 'cdos' (as ll the hard work has already been done).